ui-admin/src/portal/publish/PortalEnvDiffView.tsx

```diff
@@ -230,7 +234,14 @@ export default function PortalEnvDiffView(
     </div>
     <div className="d-flex justify-content-center mt-2 pb-5">
       { userHasPermission(user, portal.id, 'publish') && <>
-        <Button variant="primary" onClick={() => applyChanges(selectedChanges)}>
+        <Button variant="primary" onClick={() => {
+          if (isEmpty(selectedChanges.configChanges) &&
```

Comment:
why do we need this first isEmpty check?

Reply:
the first isEmpty checks for portal environment config changes, the second checks for any study environment config changes

there are a lot of isEmpty calls here as-written, so I just rewrote it slightly to hopefully be a bit more readable